Builds directly from a commit, for easier patching.

Currently synced to <https://github.com/thelounge/thelounge/commit/f5867c3643ba8ef24dcf5a7010f39fd6050d0954>

---
 .SRCINFO   |   11 +++--------
 .gitignore |    3 +++
 PKGBUILD   |   61 +++++++++++++++++++++++++++++--------------------------------
 3 files changed, 35 insertions(+), 40 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = thelounge
 	pkgdesc = Modern self-hosted web IRC client
 	pkgver = 4.4.3
-	pkgrel = 3
+	pkgrel = 3.1
 	url = https://thelounge.chat/
 	arch = any
 	license = MIT
@@ -10,19 +10,14 @@ pkgbase = thelounge
 	makedepends = python-setuptools
 	makedepends = git
 	depends = nodejs
-	noextract = thelounge-4.4.3.tgz
 	options = !lto
 	backup = etc/thelounge/config.js
-	source = https://registry.npmjs.org/thelounge/-/thelounge-4.4.3.tgz
-	source = https://raw.githubusercontent.com/thelounge/thelounge/v4.4.3/yarn.lock
-	source = https://raw.githubusercontent.com/thelounge/thelounge/v4.4.3/package.json
+	source = thelounge-f5867c3643ba8ef24dcf5a7010f39fd6050d0954.tar.gz::https://github.com/thelounge/thelounge/archive/f5867c3643ba8ef24dcf5a7010f39fd6050d0954.tar.gz
 	source = system.service
 	source = user.service
 	source = sysusers.d
 	source = tmpfiles.d
-	sha256sums = 685d753ad080c35ac7c4cf2eef6437df8de47a3d352b10f06d56acc9a08f4f18
-	sha256sums = 0bcb88814337e6dd1352dd382a14e6c277f71410767277db34c97b998e1cd0b9
-	sha256sums = e0cca2931e46842039fcbc78efbecae4b9893608d18959f616aa7ea559311441
+	sha256sums = 3cd68775f5ebb904021c67c6643ae894d727f195d26ce43bce4a01a50dd495e3
 	sha256sums = c92210f6ac8f01c1cd01b6b26793094cd2feea583ed21fab3564d6bcafdc7a20
 	sha256sums = c609f3309f54bd6285e99ff29ca2464828bec7bbbca67243ee688bd2d605dbf0
 	sha256sums = 30fab63b8a4ffcfdda4c5b8d7c66822a323c4f1de6ca62b77fe9500f4befc0a5
Index: b/.gitignore
===================================================================
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,3 @@
+*.pkg.tar.*
+*.tar.gz
+*.tgz
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,8 +2,7 @@
 # Maintainer: Maxime Poulin <maxpoulin64@gmail.com>
 pkgname=thelounge
 pkgver=4.4.3
-pkgsuffix="" #-rc.1
-pkgrel=3
+pkgrel=3.1
 pkgdesc='Modern self-hosted web IRC client'
 url='https://thelounge.chat/'
 arch=('any')
@@ -12,55 +11,53 @@ depends=('nodejs')
 options=('!lto')
 makedepends=('yarn' 'python' 'python-setuptools' 'git')
 backup=('etc/thelounge/config.js')
+_commit='f5867c3643ba8ef24dcf5a7010f39fd6050d0954'
 source=(
-    "https://registry.npmjs.org/$pkgname/-/$pkgname-${pkgver}${pkgsuffix}.tgz"
-    "https://raw.githubusercontent.com/thelounge/thelounge/v${pkgver}${pkgsuffix}/yarn.lock"
-    "https://raw.githubusercontent.com/thelounge/thelounge/v${pkgver}${pkgsuffix}/package.json"
+    "thelounge-${_commit}.tar.gz::https://github.com/thelounge/thelounge/archive/${_commit}.tar.gz"
     'system.service'
     'user.service'
     'sysusers.d'
     'tmpfiles.d'
 )
-noextract=("$pkgname-${pkgver}${pkgsuffix}.tgz")
-sha256sums=('685d753ad080c35ac7c4cf2eef6437df8de47a3d352b10f06d56acc9a08f4f18'
-            '0bcb88814337e6dd1352dd382a14e6c277f71410767277db34c97b998e1cd0b9'
-            'e0cca2931e46842039fcbc78efbecae4b9893608d18959f616aa7ea559311441'
+sha256sums=('3cd68775f5ebb904021c67c6643ae894d727f195d26ce43bce4a01a50dd495e3'
             'c92210f6ac8f01c1cd01b6b26793094cd2feea583ed21fab3564d6bcafdc7a20'
             'c609f3309f54bd6285e99ff29ca2464828bec7bbbca67243ee688bd2d605dbf0'
             '30fab63b8a4ffcfdda4c5b8d7c66822a323c4f1de6ca62b77fe9500f4befc0a5'
             'c07fc7aaa91f6d2407d9ea2d15bfa780bfc06e3487efa138a9385307dcf9f41d')
 
 prepare() {
-    yarn install --prod --frozen-lockfile --non-interactive --ignore-scripts --cache-folder "$srcdir/yarn-cache"
+	echo /etc/thelounge > "$pkgname-${_commit}/.thelounge_home"
+	cd "$pkgname-${_commit}" || exit 1
+	yarn install --frozen-lockfile --non-interactive --ignore-scripts --cache-folder "$srcdir/yarn-cache"
 }
 
 build() {
-    mkdir _build
-    cp package.json yarn.lock _build
-    cd _build || exit 1
-
-    # Install the package itself
-    # we on purpose don't use yarn global add, because --ignore-scripts
-    # is ignored: https://github.com/yarnpkg/yarn/issues/8291 but we tried
-    yarn add --no-default-rc --frozen-lockfile \
-    --prod --non-interactive --ignore-scripts \
-    --cache-folder "$srcdir/yarn-cache" --offline \
-    file:"$srcdir/$pkgname-${pkgver}${pkgsuffix}.tgz"
-
-    # fetch sqlite3 binary blob
-
-    # node-gyp or node have a bug that prevents building with "text file busy" if the kernel is too fast
-    # so we have to disable IO_URING support. This is cleary a hack and needs to be removed as soon as possible
-    # https://github.com/nodejs/node/issues/48444 is the necro bumped thread originally in docker
-    export UV_USE_IO_URING=0
-
-    cd node_modules/sqlite3 || exit 1
-    yarn run install
+	cd "$pkgname-${_commit}" || exit 1
+	export NODE_ENV=production
+	yarn build
+	mkdir _build
+	local tarball=$(npm pack --pack-destination _build)
+
+	# ensure we get the lockfile that we expect and fetch the deps
+	cp package.json yarn.lock _build
+	cd _build || exit 1
+
+	# Install the package itself
+	# we on purpose don't use yarn global add, because --ignore-scripts
+	# is ignored: https://github.com/yarnpkg/yarn/issues/8291 but we tried
+	yarn add --no-default-rc --frozen-lockfile \
+		--prod --non-interactive --ignore-scripts \
+		--cache-folder "$srcdir/yarn-cache" --offline \
+		file:"$(readlink -e "./$tarball")"
+
+	# build sqlite3 from source, avoids the binary blob (kinda)
+	cd node_modules/sqlite3 || exit 1
+	yarn --no-default-rc --frozen-lockfile --non-interactive run rebuild
 }
 
 package() {
     install -dm755 "$pkgdir/usr/lib/thelounge"
-    cp -r "$srcdir/_build/node_modules" "$pkgdir/usr/lib/thelounge"
+    cp -r "$pkgname-${_commit}/_build/node_modules" "$pkgdir/usr/lib/thelounge"
 
     install -dm755 "$pkgdir/usr/bin/"
     ln -s "/usr/lib/thelounge/node_modules/thelounge/index.js" "$pkgdir/usr/bin/thelounge"
